frame/plasm-rewards/src/lib.rs

```diff
@@ -0,0 +1,428 @@
+                start: None,
+            });
+            new_index
+        });
+
+        // let bonding_duration = T::BondingDuration::get();
+
+        BondedEras::mutate(|bonded| {
+            bonded.push((active_era, start_session));
+
+            // if active_era > bonding_duration {
+            //     let first_kept = active_era - bonding_duration;
+            //
+            //     // prune out everything that's from before the first-kept index.
+            //     let n_to_prune = bonded.iter()
+            //         .take_while(|&&(era_idx, _)| era_idx < first_kept)
+            //         .count();
+            //
+            //     // kill slashing metadata.
+            //     for (pruned_era, _) in bonded.drain(..n_to_prune) {
+            //         slashing::clear_era_metadata::<T>(pruned_era);
+            //     }
+            //
+            //     if let Some(&(_, first_session)) = bonded.first() {
+            //         T::SessionInterface::prune_historical_up_to(first_session);
+            //     }
+            // }
+        });
+    }
+
+    /// Compute payout for era.
+    fn end_era(active_era: ActiveEraInfo<MomentOf<T>>, _session_index: SessionIndex) {
+        // Note: active_era_start can be None if end era is called during genesis config.
+        if let Some(active_era_start) = active_era.start {
+            // The set of total amount of staking.
+            let now = T::Time::now();
+            let era_duration = now - active_era_start;
+
+            if !era_duration.is_zero() {
+                let total_payout = T::Currency::total_issuance();
+                let for_dapps = T::GetForDappsStaking::get_era_staking_amount(active_era.index);
+                let for_security =
+                    T::GetForSecurityStaking::get_era_staking_amount(active_era.index);
+                println!("for_security:{:?}, for_dapps:{:?}", for_security, for_dapps);
+                let (for_security_reward, for_dapps_rewards) =
+                    T::ComputeTotalPayout::compute_total_payout(
+                        total_payout,
+                        era_duration.saturated_into::<u64>(),
+                        for_security,
+                        for_dapps,
+                    );
+                println!(
+                    "for_security_reward:{:?}, for_dapps_reward:{:?}",
+                    for_security_reward, for_dapps_rewards,
+                );
+                <ForSecurityEraReward<T>>::insert(active_era.index, for_security_reward);
+                <ForDappsEraReward<T>>::insert(active_era.index, for_dapps_rewards);
+            }
+        }
+    }
+
+    /// Plan a new era. Return the potential new staking set.
+    fn new_era(start_session_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+        // Increment or set current era.
+        let current_era = CurrentEra::mutate(|s| {
+            *s = Some(s.map(|s| s + 1).unwrap_or(0));
+            s.unwrap()
```

Comment:
`unwrap` isn't safe, right?

Reply:
Before `unwrap`, the execute that:
`*s = Some(s.map(|s| s + 1).unwrap_or(0));`.

So, `unwrap` is safe.

Reply:
What’s about to use construction like

match s {
  Some(x) => x + 1,
  None => 0,
}

Instead of map, unwrap and assumption